inc/indexables/validators/class-object-type-validator.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Internals
+ */
+
+/**
+ * Class WPSEO_Object_Type_Validator
+ */
+class WPSEO_Object_Type_Validator implements WPSEO_Endpoint_Validator {
+
+	/**
+	 * Validates the object_type parameter.
+	 *
+	 * @param string $object_type The object type to validate.
+	 *
+	 * @return void
+	 *
+	 * @throws WPSEO_Invalid_Argument_Exception The invalid argument exception.
+	 */
+	private static function validate_type( $object_type ) {
+		if ( ! in_array( $object_type, array( 'post', 'term' ), true ) ) {
+			throw WPSEO_Invalid_Argument_Exception::invalid_type( 'object type' );
+		}
+
+		if ( ! WPSEO_Validator::is_string( $object_type ) ) {
+			throw WPSEO_Invalid_Argument_Exception::invalid_string_parameter( $object_type, 'object_type' );
+		}
```

Comment:
These two clauses should probably be switched around. If `$object_type` is not a string, it will already have failed either way before, because it is none of "post" / "term".